wasmcloud-test-util/src/lib.rs

```diff
@@ -4,13 +4,71 @@ pub mod provider_test;
 #[cfg(not(target_arch = "wasm32"))]
 pub mod cli;
 
-// re-export testing interface
-pub use wasmcloud_interface_testing as testing;
-
 // re-export regex and nkeys
 pub use nkeys;
 pub use regex;
 
+pub mod testing {
+
+    smithy_bindgen::smithy_bindgen!("testing/testing.smithy", "org.wasmcloud.interface.testing");
+
+    // after sdk is split we won't have to duplicate this code
+    impl Default for TestOptions {
+        fn default() -> TestOptions {
+            TestOptions {
+                patterns: vec![".*".to_string()],
+                options: std::collections::HashMap::default(),
+            }
+        }
+    }
+
+    /// A NamedResult, generated inside a `run_selected!`` or `run_selected_spawn!`` macro,
+    /// contains a tuple of a test case name and its result. The implementation of From here
+    /// makes it easy to use the `into()` function to turn the NamedResult into a TestResult.
+    pub type NamedResult<'name, T> = (&'name str, RpcResult<T>);
```

Comment:
Nit: I think I agree with @rvolosatovs here. This looks a bit wonky and would be wonky to use. I'd rather have these in a struct with some accessors (or `pub` data members) along with the `From` impl below. But I think in almost all cases we've been using `TestResult`, so keeping this as a nit